src/Symfony/Component/Security/Http/Firewall/AccessListener.php

```diff
@@ -72,12 +74,20 @@ public function authenticate(RequestEvent $event)
         $attributes = $request->attributes->get('_access_control_attributes');
         $request->attributes->remove('_access_control_attributes');
 
-        if (!$attributes || ([AuthenticatedVoter::IS_AUTHENTICATED_ANONYMOUSLY] === $attributes && $event instanceof LazyResponseEvent)) {
+        if (!$attributes || [AuthenticatedVoter::IS_AUTHENTICATED_ANONYMOUSLY] === $attributes) {
```

Comment:
I don't understand the internals of this ... but are we going to keep this `IS_AUTHENTICATED_ANONYMOUSLY`permission? I hope we can get rid of everything about "anonymous + authenticated".

Reply:
There DOES still need to be something like this, as it’s used when you are “whitelisting” paths in access_control. But it doesn’t need to have this name... though changing the name could be done later - that may simplify things.

Reply:
Yeah, so we need an attribute to whitelist pages that are within the firewall, but accessible for unauthenticated users (e.g. the login page). 

A couple suggestions after a chat about this:
1. `PUBLIC_ACCESS`
3. `ALLOW_ACCESS_TO_ALL`
2. `NO_SECURITY`

Do you, @javiereguiluz, or someone else favor one of these options (or another suggestion)?

Reply:
Also that we want to get rid of anonymous authentication the trigger word anonymous is still very good to describe an unprotected access. So another option could be ANONYMOUS_ACCESS in this context.

Reply:
Random idea ... add a `exclude` option for `access_control` entries:

```yaml
access_control:
    - { path: '^/admin/users', roles: ROLE_SUPER_ADMIN }
    - { path: '^/admin', roles: ROLE_ADMIN, exclude: ['^/admin/login', '^/admin/public/*'] }
```

Reply:
Thanks for thinking outside the box @javiereguiluz! I'm not totally sure if this would make things much clearer, given the access control is already a list of patterns. Having a sub-list of excluded pattern for each pattern might make it more confusing to see which rule is used for which request. Maybe we have to completely revisit the access control configuration in another version of Symfony :smile: 

I've updated the PR to use `PUBLIC_ACCESS`. Using anonymous is correct, it's maybe best to (at least for the next versions) avoid that term to remove confusion. Also, any logged in user is not accessing such a page "anonymously".